content/blog/2023-12-11-btor2i.md

```diff
@@ -0,0 +1,78 @@
+
+Next, we used these tests to benchmark the performance of our original interpreter versus our optimized interpreter. We show our results in the below tables; we chose to average together the results over all core benchmarks and report the results for our larger stress test separately.
+
+Below are the results for our core benchmarks. We ran three runs for each core benchmark for a given number of iterations and averaged the runtimes over all runs over all core benchmarks.
+
+| Number of Iterations  | 10             | 100            | 1000            | 10000            | 100000             | 1000000              |
+|-------------------|----------------|----------------|-----------------|------------------|--------------------|----------------------|
+| Naive interpreter | 2.0 ± 3.7 ms | 3.0 ± 3.1 ms | 11.4 ± 5.1 ms | 86.0 ± 26.0 ms | 814.3 ± 241.3 ms | 7778.9 ± 1506.2 ms |
+| Faster intepreter | 1.9 ± 3.3 ms | 1.9 ± 1.3 ms | 6.0 ± 2.9 ms  | 42.6 ± 16.4 ms | 397.8 ± 117.7 ms | 3705.3 ± 930.0 ms  |
+
+Below are the results for our stress test. We again ran three runs for the stress test for a given number of iterations and again averaged the runtimes over all runs.
+
+| Number of Trials  | 10           | 100           | 1000          | 10000           | 100000             | 1000000             |
+|-------------------|--------------|---------------|---------------|-----------------|--------------------|---------------------|
+| Naive interpreter | 2.2 ± 1.9 ms | 11.6 ± 2.9 ms | 92.8 ± 6.1 ms | 904.4 ± 33.8 ms | 10752.8 ± 180.1 ms | 53675.5 ± 326.8 ms  |
+| Faster intepreter | 2.1 ± 1.2 ms | 3.7 ± 0.9 ms  | 42.1 ± 4.0 ms | 388.7 ± 18.0 ms | 4073.9 ± 280.2 ms  | 27950.3 ± 1032.5 ms |
```

Comment:
Nice! It's really cool that your optimized heap turned out to be such a substantial win.